homeassistant/components/usb/__init__.py

```diff
@@ -61,8 +61,7 @@ def async_register_scan_request_callback(
     return discovery.async_register_scan_request_callback(callback)
 
 
-@hass_callback
-def async_is_plugged_in(hass: HomeAssistant, matcher: USBCallbackMatcher) -> bool:
+async def async_is_plugged_in(hass: HomeAssistant, matcher: USBCallbackMatcher) -> bool:
```

Comment:
I don't think that we should convert this to async just because 1 call needs to have a scan included. 

Instead, we should create a 2nd function `async_check_is_plugged_in` that does a scan + returns `asnyc_is_plugged_in`.

Reply:
I don't agree, `async_is_plugged_in` was added specifically for sky_connect, and is not used by anything else.
Without waiting for a scan, `async_is_plugged_in` unconditionally returns `False` during Home Assistant startup which maked the function useless.

Reply:
Right, but I suggest to make a wrapper function that does scan + is_plugged_in, so we could keep is_plugged_in the same for places like `async_info`, which is now turning async as a side-effect.